department-of-veterans-affairs/va.gov-cms
# Add event subscriber to include situation updates for full banner width alerts

## Description

closes #16781 

## Testing done

Manually tested as well as added/updated integration tests.

## Screenshots

example output at "api/v1/banner-alerts".

<img width="479" alt="Screenshot 2024-01-30 at 11 16 11 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3640707/74cdc1fe-fe61-40ee-ae1f-32373e524b24">

## QA steps

1. Edit a "full_width_banner_alert" node
   - [x] Add two situation updates to that node
   - [x] Figure out which path it loads on via the associated VAMC systems
2. Add a "banner" node that loads at the same path as the "full_width_banner_alert" node you just edited.
3. Add a "promo_banner" node that loads at the same path as the "full_width_banner_alert" node you just edited.
4. Go to `/api/v1/banner-alerts?path=/your-path`
   - [x] Confirm that the "banner" data loads correctly
   - [x] Confirm that the "promo_banner" data loads correctly
   - [x] Confirm that the "full_width_banner_alert" data loads correctly and includes "field_situation_updates" paragraph data either in the "attributes" or "includes" section of the response.
 5. Try not adding a path, `/api/v1/banner-alerts`, and confirm the response is `error: "No path provided."`.
 6. Try a path that shouldn't return any data `/api/v1/banner-alerts?path=/does-not-exist` and confirm that no data is returned. Make sure to check if a promo/banner is set to `*` and shows on all paths. This might confuse your test since that banner will show even on paths Drupal thinks are 404s...and maybe this should be updated, but it wasn't in the scope of this ticket. 

### Definition of Done


### Select Team for PR review
